optimum/habana/transformers/models/clip/modeling_clip.py

```diff
@@ -16,3 +34,310 @@ def forward(self, pixel_values: torch.FloatTensor) -> torch.Tensor:
+        self._hpu_kernel_fsdpa = fusedSDPA
+
+    def forward(self, query, key, value, attn_mask, dropout_p, is_casual, scale, softmax_mode):
+        return self._hpu_kernel_fsdpa.apply(query, key, value, attn_mask, dropout_p, is_casual, scale, softmax_mode)
+
+
+class Matmul(torch.nn.Module):
+    def __init__(self):
+        super().__init__()
+
+    def forward(self, x, y):
+        return torch.matmul(x, y)
+
+
+class Softmax(nn.Module):
+    def __init__(self):
+        super().__init__()
+
+    def forward(self, x, dim=None, invAttnHead=None):
+        return torch.ops.hpu.softmax_fp8(x, dim, None, None, invAttnHead)
+
+
+class GaudiCLIPAttention(CLIPAttention):
+
+    def __init__(self, config):
+        super().__init__(config=config)
+        self.fused_scaled_dot_product_attention = ModuleFusedSDPA(FusedSDPA) if FusedSDPA else None
+        self.bmm1 = Matmul()
+        self.bmm2 = Matmul()
+        self.softmax = Softmax()
+
+    def forward(
+        self,
+        hidden_states: torch.Tensor,
+        attention_mask: Optional[torch.Tensor] = None,
+        causal_attention_mask: Optional[torch.Tensor] = None,
+        output_attentions: Optional[bool] = False,
+        use_flash_attention: Optional[bool] = False,
+    ) -> Tuple[torch.Tensor, Optional[torch.Tensor], Optional[Tuple[torch.Tensor]]]:
+        """
+        Copied from CLIPAttention.forward: https://github.com/huggingface/transformers/blob/ab0f050b42d903f34d6eb97f3f8c0c07f0517ad2/src/transformers/models/clip/modeling_clip.py
+        The only differences are:
+        - add new args use_flash_attention
+        """
+        bsz, tgt_len, embed_dim = hidden_states.size()
+        attn_weights_reshaped = None
+        # get query proj
+        query_states = self.q_proj(hidden_states) * self.scale
+        key_states = self._shape(self.k_proj(hidden_states), -1, bsz)
+        value_states = self._shape(self.v_proj(hidden_states), -1, bsz)
+
+        proj_shape = (bsz * self.num_heads, -1, self.head_dim)
+        query_states = self._shape(query_states, tgt_len, bsz).view(*proj_shape)
+        key_states = key_states.view(*proj_shape)
+        value_states = value_states.view(*proj_shape)
+
+        src_len = key_states.size(1)
+        if FusedSDPA and use_flash_attention:
+            import habana_frameworks.torch.hpu as ht
+
+            if tgt_len == 1 or causal_attention_mask is None:
+                with ht.sdp_kernel(enable_recompute=True):
+                    attn_output = self.fused_scaled_dot_product_attention(
+                        query_states, key_states, value_states, attention_mask, self.dropout, False, 1, "fast"
+                    )
+            elif causal_attention_mask is not None:
+                with ht.sdp_kernel(enable_recompute=True):
+                    attn_output = self.fused_scaled_dot_product_attention(
+                        query_states, key_states, value_states, causal_attention_mask, self.dropout, True, 1, "fast"
```

Comment:
are you sure about using causal_attention_mask as an input to the fused sdpa?
from other examples with is_causal=true, None is passed instead.